crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -4615,16 +4615,17 @@ impl<'db> TypeInferenceBuilder<'db> {
                     .unwrap_or_else(|| KnownClass::Int.to_instance(self.db())),
             ),
 
-            (Type::IntLiteral(n), Type::IntLiteral(m), ast::Operator::Pow) => {
-                let m = u32::try_from(m);
-                Some(match m {
-                    Ok(m) => n
-                        .checked_pow(m)
+            (Type::IntLiteral(n), Type::IntLiteral(m), ast::Operator::Pow) => Some({
+                if m < 0 {
+                    KnownClass::Float.to_instance(self.db())
```

Comment:
Would we fall back to `float` anyway via typeshed stubs if we guard the whole match pattern with `m < 0`? Or do we not understand the typeshed stubs yet? Either way, I'm also fine with hardcoding `float` here.

Reply:
I doubt there are many code bases that have so many pow operations that the performance difference is noticeable 😉